frontend/models/product/server.ts

```diff
@@ -53,24 +46,31 @@ export async function findProduct({
          ),
       ]);
 
-   if (iFixitQueryResponse?.redirectSkuUrl) {
+   return (
+      getIFixitRedirect() ??
+      (await getProduct({
+         shopifyProduct: shopifyQueryResponse.product,
+         strapiProduct: strapiQueryResponse.products?.data[0],
+         iFixitProduct: iFixitQueryResponse,
+      })) ??
+      getShopifyRedirect()
```

Comment:
Yes, that's my understanding.

If there is an iFixit redirect from product A to B, then our admins always want A to redirect to B. Therefore it takes precedence.

Our backend records Shopify product redirects when a product's handle is changed (say it was changed from X to Y). So if a user visits `/products/X`, instead of 404ing they will redirect to `/products/Y` which has content. If an admin later creates a new product with handle `X`, then we want to serve content for `/products/X` instead of redirecting. Therefore, we will ignore the redirect and the product query will take precedence.

---

Another observation about Shopify redirects: On Shopify themes, Shopify redirects only apply if the `path` does not exist. So our implementation aligns with Shopify's theme behavior.

For example, on our DE store there happens to be a redirect from `/collections` to `/` configured.

![image](https://github.com/iFixit/react-commerce/assets/52104630/6798f0fb-32a2-4473-9b8a-b95e6c5272c7)
https://admin.shopify.com/store/ifixit-de/redirects.json

However, when you visit https://store.ifixit.de/collections, there is no redirect. The resource exists, and takes precedence over the redirect.

Reply:
I see, makes sense 👍